### DISCUSSION TITLE
Alert by insertion time instead of @timestamp

### DISCUSSION THREAD

**Discussion by shaybuhadana** [Security-Onion-Solutions/securityonion#6196] [Unsupported Versions]
Hi!
We are using Securityonion 2.3.60 in distribute model.
We encounter a problem when the data comes in delay (because large scale of data and faults).
The delay can be up to 12 hours.
In that case, the elastalert cannot alert over the delayed data because it searches over the field @timesatmp.
Those missing alerts are very important for us.

Is anyone encountered with this case? 
Or got an idea for solving this issue?

Thank you very much!

**Comment by abesinger** [comment]
Not a complete answer, but you might try alerting on `event.created` instead of `@timestamp`.  It'll depend on where your latency is -- `event.created` is set to the time that the event is processed by *beat (assuming you're using a beat). If your processing delay is happening post-beat it won't help you.

It could be helpful to have a field like `event.ingested` added at the _end_ of the ingest pipeline, but that's not something available today.

**Reply by shaybuhadana** [reply]
Thank you for your answer!
Its seems that we need to add `event.ingested` in the elasticsearch pipelines. 
But for the elastalert side, is there an easy way to configure the alert time field to be `event.ingested` instead of `@timesatmp`?

**Reply by abesinger** [reply]
sorry, can't help with that part.

**Comment by dougburks** [comment]
> We encounter a problem when the data comes in delay (because large scale of data and faults).
> The delay can be up to 12 hours.

I would recommend focusing on this problem first. Seems like your deployment may be undersized for the amount of data you are ingesting. Double-check that your hardware meets the recommendations here:
https://docs.securityonion.net/en/2.3/hardware.html

**Reply by shaybuhadana** [reply]
Hi!
The problem of the delay we are dealing with is not in the Security Onion side.
We setup our environment as the hardware recommendations you mentioned.
Thank you very much!
